lib/src/Uncertainty/Distribution/PointConditionalDistribution.cxx

```diff
@@ -0,0 +1,648 @@
+        R(i, j) = C(i, j) / (sigma[i] * sigma[j]);
+    }
+    simplified = Student(nu + py, mu, sigma, R);
+    return true;
+  }
+
+  // JointDistribution
+  JointDistribution *p_joint = dynamic_cast<JointDistribution *>(distribution_.getImplementation().get());
+  if (p_joint && p_joint->getCore().isCopula() && 0)
+  {
+    const JointDistribution::DistributionCollection collection(p_joint->getDistributionCollection());
+    JointDistribution::DistributionCollection marginalCollection(getDimension());
+    for (UnsignedInteger i = 0; i < getDimension(); ++i)
+      marginalCollection[i] = collection[nonConditioningIndices_[i]];
+    Point coreConditioningValues(conditioningValues_.getDimension());
+    for (UnsignedInteger i = 0; i < conditioningValues_.getDimension(); ++ i)
+      coreConditioningValues[i] = collection[conditioningIndices_[i]].computeCDF(conditioningValues_[i]);
+    const PointConditionalDistribution conditionedCore(p_joint->getCore(), conditioningIndices_, coreConditioningValues);
+    simplified = JointDistribution(marginalCollection, conditionedCore.getSimplifiedVersion());
+    return true;
+  }
+
+  return false;
+}
+
+Distribution PointConditionalDistribution::getSimplifiedVersion() const
+{
+  return useSimplifiedVersion_ ? simplifiedVersion_ : *this;
+}
+
+/* Compute the numerical range of the distribution given the parameters values */
+void PointConditionalDistribution::computeRange()
+{
+  if (useSimplifiedVersion_)
+    setRange(simplifiedVersion_.getRange());
+
+  setRange(marginalConditionedDistribution_.getRange());
+}
+
+/* Get the support of a distribution that intersect a given interval */
+Sample PointConditionalDistribution::getSupport(const Interval & interval) const
+{
+  if (interval.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given interval has a dimension that does not match the distribution dimension.";
+  if (!isDiscrete()) throw NotDefinedException(HERE) << "Error: the support is defined only for discrete distributions.";
+  Collection<Sample> marginalSupport(getDimension());
+  Indices marginalSize(getDimension());
+  for (UnsignedInteger j = 0; j < getDimension(); ++ j)
+  {
+    marginalSupport[j] = marginalConditionedDistribution_.getMarginal(j).getSupport(interval.getMarginal(j));
+    marginalSize[j] = marginalSupport[j].getSize();
+  }
+  IndicesCollection tuples(Tuples(marginalSize).generate());
+  Sample support(0, getDimension());
+  for (UnsignedInteger i = 0; i < tuples.getSize(); ++ i)
+  {
+    Point x(getDimension());
+    for (UnsignedInteger j = 0; j < getDimension(); ++ j)
+      x[j] = marginalSupport[j](tuples(i, j), 0);
+    support.add(x);
+  }
+  return support;
+}
+
+/* Get one realization of the distribution */
+Point PointConditionalDistribution::getRealization() const
+{
+  if (useSimplifiedVersion_)
+    return simplifiedVersion_.getRealization();
+
+  if (distribution_.hasIndependentCopula())
```

Comment:
if distribution_ has an independent copula, then it has a simplified version which is simply its marginal wrt the nonconditioning indices